script/validate-categories/validate.rb

```diff
@@ -0,0 +1,37 @@
+require 'json'
+
+settings = JSON.parse(File.read('./settings.json'))
+folders = settings['folders']
+@allowed_categories = settings['allowed_categories']
+
+def validateCategories(categories)
+    return categories.nil? || (categories.is_a?(Array) && ! categories.select{|l| @allowed_categories.detect{|p| p.casecmp(l) == 0 } }.empty? )
```

Comment:
If categories is nil, then validate should fail

Reply:
categories is an optional field. We do have a blank template today where there are no categories.

Reply:
This seems troublesome to me. Blank is a special case we should handle is separately. For any other template, we can disallow missing `categories`.